Mathlib/MeasureTheory/Constructions/Polish.lean

```diff
@@ -58,13 +58,86 @@ We use this to prove several versions of the Borel isomorphism theorem.
 
 open Set Function PolishSpace PiNat TopologicalSpace Metric Filter Topology MeasureTheory
 
-variable {α : Type*} [TopologicalSpace α] {ι : Type*}
+/-! ### Standard Borel Spaces -/
+
+variable (α : Type*)
+
+/-- A standard Borel space is a measurable space arising as the borel sets of some Polish topology.
+This is useful in situations where a space has no natural topology or
+the natural topology in a space is non-Polish.
+
+To endow a standard Borel space `α` with a compatible Polish topology, use
+`letI := upgradePolishSpace α`. One can then use `eq_borel_upgradeStandardBorel α` to
+rewrite the `MeasurableSpace α` instance to `borel α t`, where `t` is the new topology.-/
+class StandardBorelSpace [MeasurableSpace α] : Prop where
+  /-- There exists a compatible Polish topology. -/
+  polish : ∃ _ : TopologicalSpace α, BorelSpace α ∧ PolishSpace α
+
+/-- A convenience class similar to `UpgradedPolishSpace`. No instance should be registered.
+Instead one should use `letI := upgradePolishSpace α`. -/
+class UpgradedStandardBorel extends MeasurableSpace α, TopologicalSpace α,
+  BorelSpace α, PolishSpace α
+
+/-- Use as `letI := upgradePolishSpace α` to endow a standard Borel space `α` with
+a compatible Polish topology.
+
+Warning: following this with `borelize α` will cause an error. Instead, one can
+rewrite with `eq_borel_upgradeStandardBorel α`. -/
+noncomputable
+def upgradeStandardBorel [MeasurableSpace α] [h : StandardBorelSpace α] :
+    UpgradedStandardBorel α := by
+  choose τ hb hp using h.polish
+  constructor
+
+/-- The `MeasurableSpace α` instance on a `StandardBorelSpace` `α` is equal to
+the borel sets of `upgradeStandardBorel α`.-/
+theorem eq_borel_upgradeStandardBorel [MeasurableSpace α] [StandardBorelSpace α] :
+    ‹MeasurableSpace α› = @borel _ (upgradeStandardBorel α).toTopologicalSpace :=
+  @BorelSpace.measurable_eq _ (upgradeStandardBorel α).toTopologicalSpace _
+    (upgradeStandardBorel α).toBorelSpace
+
+variable {α} [MeasurableSpace α]
```

Comment:
from this point on, `[MeasurableSpace α]` is active for the rest of the file, while it is not relevant later on. Could you open a `section` and only activate it inside this section?